scripts/dropins.js

```diff
@@ -12,15 +12,84 @@ import { initializers, Initializer } from '@dropins/tools/initializer.js';
 // Drop-ins
 import * as authApi from '@dropins/storefront-auth/api.js';
 import * as cartApi from '@dropins/storefront-cart/api.js';
+import * as orderApi from '@dropins/storefront-order/api.js';
 
 // Recaptcha
 import * as recaptcha from '@dropins/tools/recaptcha.js';
 
 // Libs
 import { getConfigValue, getCookie } from './configs.js';
+import { getMetadata } from './aem.js';
 
 export const getUserTokenCookie = () => getCookie('auth_dropin_user_token');
 
+const initializeOrderApi = (orderRef) => {
+  initializers.register(orderApi.initialize, {
+    orderRef,
+  });
+};
+
+const redirectTo = (path) => {
+  window.location.href = path;
+};
+
+const handleUserOrdersRedirects = () => {
+  const currentUrl = new URL(window.location.href);
+  const isAccountPage = currentUrl.pathname.includes('/customer');
+  const isAuthenticated = !!getCookie('auth_dropin_user_token') ?? false;
+  const orderRef = currentUrl.searchParams.get('orderRef');
+  const isTokenProvided = orderRef && orderRef.length > 20;
+
+  const ORDER_STATUS_PATH = '/order-status';
+  const ORDER_DETAILS_PATH = '/order-details';
+  const CUSTOMER_ORDER_DETAILS_PATH = '/customer/order-details';
+  const CUSTOMER_ORDERS_PATH = '/customer/orders';
+  const ORDER_REF_URL_QUERY = `?orderRef=${orderRef}`;
+
+  let targetPath = null;
+  if (currentUrl.pathname.includes(CUSTOMER_ORDERS_PATH)) {
+    return;
+  }
+
+  events.on('order/error', ({ error }) => {
+    const defaultErrorMessage = "We couldn't locate an order with the information provided.";
```

Comment:
You should get this from the dropin, rather than re-defining here, otherwise we have to always keep this and the dropin message in sync for the below conditional to be valid.